packages/bp/src/orchestrator/migrator.ts

```diff
@@ -0,0 +1,36 @@
+import { createLoggerProvider } from 'core/app/core-loader'
+import versions from 'core/migration/versions.json'
+import { spawnWebWorker, startNluServer, startStudio } from 'orchestrator'
+import { startMessagingServer } from './messaging-server'
+
+/**
+ * When the command migrate up or down is used, we start each process with the migration arguments.
+ * They must handle all arguments and exit automatically when finished
+ */
+export const runMigrator = async () => {
+  const loggerProvider = createLoggerProvider()
+  const logger = await loggerProvider('[Migrator]')
+
+  logger.info('Starting Messaging migrations...')
+  await Promise.fromCallback(cb => startMessagingServer(logger, undefined, cb))
```

Comment:
Since Botpress migrates from one known version to another, we should make a check to see (in the version.json file) if there was a version bump of each binary before starting them

Reply:
Right now the studio doesn't support down migrations, neither the nlu or messaging (for the time being). We can adjust the mechanism when we do have migrations on other processes.

Reply:
My comment is still valid for up migrations too. Let's say you are upgrading from Botpress v12.26.8 to v12.26.9 and there we no changes in the messaging server, you don't want to start a subprocess for nothing

Reply:
Yes knowing if there are migrations to run could also be useful for the auto-migrate